agent/agent.go

```diff
@@ -70,14 +77,15 @@ Starting BackupsHQ agent
 	a.logger.Info(fmt.Sprintf(`Authenticated as principal %s "%s"`, a.principal.ID, a.principal.Name))
 	a.account = a.apiClient.GetAccount(tokenInfo.AccountId)
 	a.logger.Info(fmt.Sprintf(`This agent belongs to account %s "%s"`, a.account.ID, a.account.Name))
+	a.token = a.apiClient.Register()
 
-	a.logger.Info("Sync frequency: " + a.syncFrequency)
-	a.update()
-	cr := cron.New()
-	cr.AddFunc(a.syncFrequency, func() {
-		a.update()
-	})
-	cr.Start()
+	wg.Add(1)
```

Comment:
I think this is missing:

```suggestion
	var wg sync.WaitGroup
	wg.Add(1)
```

Reply:
Duh, yep! When I added it locally (and forgot to commit it) I added it globally in this file, the reason being that we might want to interact with it inside the ping and update functions to exit gracefully, passing it in doesn't seem quite right. Does that sound correct?

Reply:
> might want to interact with it inside the ping and update functions to exit gracefully

In that case why not add it to the `Agent` struct? Then it'll be accessible in `ping` and `update`. Whatever way works for now though!

Reply:
Good idea, done!

Reply:
Nice one!